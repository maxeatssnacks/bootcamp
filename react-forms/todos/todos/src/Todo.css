.Todo {
    color: black;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 10px;
    margin-bottom: 10px;
    background-color: #f9f9f9;
    border-radius: 4px;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}

.Todo-task {
    flex-grow: 1;
}

.Todo-remove {
    background-color: #f44336;
    color: white;
    border: none;
    padding: 10px 15px;
    border-radius: 4px;
    cursor: pointer;
    transition: background-color 0.3s;
}

.Todo-remove:hover {
    background-color: #d32f2f;
}