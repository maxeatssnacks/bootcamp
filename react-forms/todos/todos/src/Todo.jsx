import React from 'react';
import "./Todo.css";

const Todo = ({ id, task, removeTodo }) => {
    return (
        <li className="Todo">
            <span className="Todo-task">{task}</span>
            <button className="Todo-remove" onClick={() => removeTodo(id)}>Delete</button>
        </li>
    );
};

export default Todo;