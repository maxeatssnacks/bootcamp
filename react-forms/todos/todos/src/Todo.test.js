import React from 'react';
import '@testing-library/jest-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import Todo from './Todo';

// Smoke test
test('Todo renders without crashing', () => {
    render(<Todo id="1" task="Test todo" removeTodo={() => { }} />);
});

// Snapshot test
test('Todo matches snapshot', () => {
    const { asFragment } = render(<Todo id="1" task="Test todo" removeTodo={() => { }} />);
    expect(asFragment()).toMatchSnapshot();
});

// Critical business logic test
test('calls removeTodo when delete button is clicked', () => {
    const mockRemoveTodo = jest.fn();
    render(<Todo id="1" task="Test todo" removeTodo={mockRemoveTodo} />);

    // Find and click the delete button
    const deleteButton = screen.getByText('Delete');
    fireEvent.click(deleteButton);

    // Check if removeTodo was called with the correct id
    expect(mockRemoveTodo).toHaveBeenCalledWith('1');
});