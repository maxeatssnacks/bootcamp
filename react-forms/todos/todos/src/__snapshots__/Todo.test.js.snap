// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="Todo"
  >
    <span
      class="Todo-task"
    >
      Test todo
    </span>
    <button
      class="Todo-remove"
    >
      Delete
    </button>
  </li>
</DocumentFragment>
`;
